packages/SwingSet/src/kernel/virtualObjectManager.js

```diff
@@ -0,0 +1,328 @@
+import { assert, details, q } from '@agoric/assert';
+import { parseVatSlot } from '../parseVatSlots';
+
+/**
+ * Make a simple LRU cache of virtual object inner selves.
+ *
+ * @param {number} size  Maximum number of entries to keep in the cache before
+ *    starting to throw them away.
+ * @param {(instanceKey: string) => Object} fetch  Function to retrieve an
+ *    object's raw state from the store by its instanceKey
+ * @param {(instanceKey: string, rawData: Object) => void} store  Function to
+ *   store raw object state by its instanceKey
+ *
+ * @returns {Object}  An LRU cache of (up to) the given size
+ */
+function makeCache(size, fetch, store) {
+  let lruHead;
+  let lruTail;
+  let count = 0;
```

Comment:
This seems like an improvement.  It was not quite as straightforward as you intuited because of a minor order of operations thing (not semantically significant) that broke one of my tests, but it was easy to fix.